File: composeApp/src/androidMain/kotlin/dem/dev/timeflame/feature/main/state/MainScreenState.kt
package dem.dev.timeflame.feature.main.state

import dem.dev.timeflame.domain.model.Task
import dem.dev.timeflame.domain.model.User
import dem.dev.timeflame.feature.calendar.model.CalendarDay
import dem.dev.timeflame.feature.calendar.model.Month
import dem.dev.timeflame.util.datetime.KDateTime
import dem.dev.timeflame.util.datetime.now
import dem.dev.timeflame.util.state.ScreenState

data class MainScreenState(
    val screenState: ScreenState = ScreenState.Idle,
    val currentMonth: Month? = null,
    val currentUser: User? = null,
    val selectedDayIndex: Int = 0,
    val voiceInputState: VoiceInputState = VoiceInputState(),
    val selectedDay: CalendarDay = CalendarDay(KDateTime.now(), mutableListOf()),
    val calendarViewState: CalendarViewState = CalendarViewState.MONTH,
    val selectedTaskToEdit: Task? = null,
    val profileContextMenuOpened: Boolean = false,
    val logoutRequested: Boolean = false
)

File: iosApp/iosApp/feature/main/MainScreenState.swift
//
//  MainScreenState.swift
//  iosApp
//
//  Created by Mikhail Maevskii on 15/12/2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import Shared

struct MainScreenState {
    var screenState: ScreenState = ScreenState.idle
    var currentMonth: Month? = Month.Companion().current()
    var selectedDay: CalendarDay = CalendarDay(day: KDateTime.Companion().now(), tasks: [])
    var calendarViewState: CalendarViewState = CalendarViewState.Month
    var navigateTo: Router.Destination? = nil
    var selectedTaskToEdit: Task? = nil
    
    func copy(
        screenState: ScreenState? = nil,
        currentMonth: Month? = nil,
        selectedDay: CalendarDay? = nil,
        calendarViewState: CalendarViewState? = nil,
        navigateTo: Router.Destination? = nil,
        selectedTaskToEdit: Task? = nil
    ) -> Self {
        return MainScreenState(
            screenState: screenState ?? self.screenState,
            currentMonth: currentMonth ?? self.currentMonth,
            selectedDay: selectedDay ?? self.selectedDay,
            calendarViewState: calendarViewState ?? self.calendarViewState,
            navigateTo: navigateTo ?? self.navigateTo,
            selectedTaskToEdit: selectedTaskToEdit ?? self.selectedTaskToEdit
        )
    }
}
